src/mbgl/map/transform.cpp

```diff
@@ -595,12 +595,16 @@ void Transform::startTransition(const CameraOptions& camera,
             }
             observer.onCameraIsChanging();
         } else {
-            transitionFinishFn();
+            auto finish = transitionFinishFn;
+
             transitionFinishFn = nullptr;
 
             // This callback gets destroyed here,
             // we can only return after this point.
```

Comment:
I think this comment was meant to serve as a reminder not to change this lambda such that any code executes "after this point". But this change does that. Is it safe? If so, please update this comment to explain why.